src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAssignableTypeForForeachVariableTests.cs

```diff
@@ -0,0 +1,533 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Runtime.CSharpUseAssignableTypeForForeachVariable,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseAssignableTypeForForeachVariableTests
+    {
+        [Fact]
+        public async Task SameType()
+        {
+            var test = @"
+using System.Collections.Generic;
+
+namespace ConsoleApplication1
+{
+    class Program
+    {   
+        void Main()
+        {
+            var x = new List<string>();
+            foreach (string item in x)
+            {
+            }
+        }
+    }
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(test, test);
```

Comment:
You have no code fix so you could use the `VerifyAnalyzerAsync` method instead but I am fine with keeping the current code.

Reply:
@Evangelink Using VerifyCodeFixAsync is better in case there is a codefix in the future.

Reply:
That's a good remark, maybe we could (in a separate PR) ban the `VerifyAnalyzerAsync` and replace all calls. I am not sure which quantity of work that represents.